### PR TITLE
Update definition of ApplicationGatewayOnDemandProbe

### PR DESCRIPTION
**Pull request by patoda**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by anton-evseev — APPROVED** [review]

**Comment by number213** [comment]
@lirenhe could you please take a look?

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json [hunk 1]
```diff
@@ -801,13 +801,13 @@
           "$ref": "#/definitions/ApplicationGatewayProbeHealthResponseMatch",
           "description": "Criterion for classifying a healthy probe response."
         },
-        "backendPoolName": {
-          "type": "string",
-          "description": "Name of backend pool of application gateway to which probe request will be sent."
+        "backendAddressPool": {
```
Comment (yungezz) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json]: this is a breaking change in stable version, pls create a new version

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/applicationGateway.json [hunk 2]
```diff
@@ -804,13 +804,13 @@
           "$ref": "#/definitions/ApplicationGatewayProbeHealthResponseMatch",
           "description": "Criterion for classifying a healthy probe response."
         },
-        "backendPoolName": {
-          "type": "string",
-          "description": "Name of backend pool of application gateway to which probe request will be sent."
+        "backendAddressPool": {
```
Comment (yungezz) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/applicationGateway.json]: breaking change in stable version

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json (hunk 1 above)
Comment (patoda) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json]: @yungezz , this should not be breaking for any customer. Reason is that previously, these changes got merged incorrectly before the NRP API changes were checked in - so no customer can use it. The NRP API (the only version) has gone in NRP/124. So, we need to correct the definitions here as per the actual backend API.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/applicationGateway.json (hunk 2 above)
Comment (patoda) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/applicationGateway.json]: @yungezz , this should not be breaking for any customer. Reason is that previously, these changes got merged incorrectly before the NRP API changes were checked in - so no customer can use it. The NRP API (the only version) has gone in NRP/124. So, we need to correct the definitions here as per the actual backend API.

**Comment by number213** [comment]
@Azure/arm-api-review-board could you please review this PR?

**Review by lirenhe — APPROVED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json (hunk 1 above)
Comment (ryansbenson) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json]: Are you saying the properties in the repo (backendPoolName...) were never supported? This change is to align the documentation/sdk with what the backend supports.

Or are you saying you've changed your backend and need to update the documentation/sdk to represent this change?

Comment (patoda) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json]: The first option. backendPoolName.. and backendHttpSettingName were never supported. This change is to align the documentation/sdk with what the backend supports.

**Review by ryansbenson — APPROVED** [review]
Approved from ARMs side

**Review by sergey-shandar — APPROVED** [review]

**Comment by number213** [comment]
@lirenhe could you please merge this PR?

**Pull request merged**